### PR TITLE
Add generator scalable modification

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/translations/network-modifications-en.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "network_modifications/TWO_WINDINGS_TRANSFORMER_CREATION": "Creating 2 windings transformer {computedLabel}",
     "network_modifications/BRANCH_STATUS": "{action, select, TRIP {Trip line {computedLabel}} LOCKOUT {Lock out line {computedLabel}} ENERGISE_END_ONE {Energise {computedLabel} on {energizedEnd}} ENERGISE_END_TWO {Energise {computedLabel} on {energizedEnd}} SWITCH_ON {Switch on {computedLabel}} other {Modification line status {computedLabel}}}",
     "network_modifications/SHUNT_COMPENSATOR_CREATION": "Creating shunt compensator {computedLabel}",
+    "network_modifications/GENERATOR_SCALING": "Creating generator scalable {computedLabel}",
```
Comment (FranckLecuyer) [inline src/translations/network-modifications-en.json:19]: Creating generator scaling {computedLabel}

File: src/translations/fr.json [hunk 2]
```diff
@@ -538,6 +538,19 @@
...
+    "TargetPText": "P cible",
+    "IterativeLabel": "Itérative",
+    "VariationMode": "Mode de variation",
+    "ProportionalToPMax": "Proportionnel à Pmax",
+    "RegularDistribution": "Équipartition",
+    "StackingUp": "Empilement",
+    "variations": "Variations",
+    "CreateVariation": "Ajouter variation",
+    "Variations": "Variations",
+    "GeneratorScalingError" : "Erreur lors de la création de Variation plan de consommation",
```
Comment (FranckLecuyer) [inline src/translations/fr.json:552]: Erreur lors de la création de la variation du plan de production

File: src/translations/en.json [hunk 3]
```diff
@@ -538,6 +538,18 @@
     "PhaseShiftValuesError": "The values of the phase tap shifts must be ordered and without duplicates",
     "TapPositionValueError": "The number of taps must not exceed the value {value}",
 
+    "GeneratorScaling": "Generator scalable",
```
Comment (FranckLecuyer) [inline src/translations/en.json:541]: : "Generator scaling"

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/components/graph/menus/modification-list-item.js [hunk 4]
```diff
@@ -73,6 +73,8 @@ export const ModificationListItem = ({
             return modif.lineToAttachToId;
         } else if (modif.type === 'LINES_ATTACH_TO_SPLIT_LINES') {
             return modif.attachedLineId;
+        } else if (modif.type === 'GENERATOR_SCALING') {
+            return modif.uuid;
```
Comment (FranckLecuyer) [inline src/components/graph/menus/modification-list-item.js:77]: displaying the modification uuid is useless for the user ... 
Maybe we should set a name in the modification creation dialog ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
Tests: 
'Iterative' mode change is not persistent (when reopening the dialog, it is always checked, even if it was previously unchecked by the user
'Iterative' mode is always 'false' in the database, even if it was checked by the user
Adding or deleting a filter does not enabling the 'Validate' button
'Stacking up' mode with 'target P' mode does not function (see remarks on back concerning the sum value not calculated)

**Review by FranckLecuyer — APPROVED** [review]

File: src/utils/rest-api.js [hunk 5]
```diff
@@ -1389,6 +1389,49 @@ export function switchOnLine(studyUuid, currentNodeUuid, lineId) {
...
+        generatorScalingUrl =
+            getStudyUrlWithNodeUuid(studyUuid, currentNodeUuid) +
+            '/network-modifications/' +
+            encodeURIComponent(modificationUuid);
+    } else {
+        console.info('create generator scaling', body);
+        generatorScalingUrl =
+            getStudyUrlWithNodeUuid(studyUuid, currentNodeUuid) +
+            '/network-modifications';
+    }
```
Comment (SlimaneAmar) [inline src/utils/rest-api.js:1419]: let generatorScalingUrl = getStudyUrlWithNodeUuid(studyUuid, currentNodeUuid) +'/network-modifications';
if (modificationUuid) {
    generatorScalingUrl = generatorScalingUrl + '/' + encodeURIComponent(modificationUuid);
}

File: src/translations/fr.json [hunk 6]
```diff
@@ -645,6 +658,7 @@
     "ReactivePowerSetpointMVAR": "Consigne de reactif (MVAR)",
     "EmptyList/bbs": "La liste de jeux de barres est vide",
     "EmptyList/connections": "La liste de connexions est vide",
+    "EmptyList/variations": "La liste de variations est vide",
```
Comment (SlimaneAmar) [inline src/translations/fr.json:661]: La liste de**s** variations est vide

**Pull request merged**